### PR TITLE
authz: configure authz granter in transfers page

### LINKED ISSUE #473: configure authz granter in each page
# Summary
If user switches to the authz mode, ask him select granter account if there are any. Check governance page for reference
* ~Transfers Page~ #516
* ~Governance Page~
* ~Staking Page(delegate, undelegate, redelegate and withdraw rewards).~ #527 
* Group page
* Feegrant page #566

### PR DESCRIPTION
**Pull request by Hemanthghs**
Added Authz granter for transfers page

### CODE REVIEW

File: frontend/src/pages/SendPage.jsx [hunk 1]
```diff
@@ -25,7 +25,31 @@ import Box from "@mui/material/Box";
...
+          "/cosmos.bank.v1beta1.SendAuthorization"
+      ) {
+        granters.push(grant.granter);
+      }
+    }
+    result[chainID] = granters;
+  }
+
+  return result;
+};
```
Comment (aleem1314) [inline frontend/src/pages/SendPage.jsx:52]:
```suggestion
export const filterSendAuthz = (authzs) => {
  const result = {};
  
  for (const chainID in authzs) {
    const granters = [];
    const grants = authzs[chainID]?.grants || [];

    for (const grant of grants) {
      const authorizationType = grant?.authorization["@type"];
      const isGenericAuthorization = authorizationType === "/cosmos.authz.v1beta1.GenericAuthorization";
      const isSendAuthorization = authorizationType === "/cosmos.bank.v1beta1.SendAuthorization";
      const isMsgSend = grant?.authorization.msg === "/cosmos.bank.v1beta1.MsgSend";

      if (isGenericAuthorization && isMsgSend || isSendAuthorization) {
        granters.push(grant.granter);
      }
    }

    result[chainID] = granters;
  }

  return result;
};
```

File: frontend/src/features/authz/authzSlice.js [hunk 2]
```diff
@@ -447,24 +447,35 @@ export const authzSlice = createSlice({
...
+          state.grantsToMe[action.payload.chainID] = result;
         }
-        state.grantsToMe[action.payload.chainID] = result;
       })
       .addCase(getGrantsToMe.rejected, (state, action) => {
-        state.grantsToMe.status = "rejected";
-        state.grantsToMe.grants = [];
-        state.grantsToMe.pagination = {};
-        state.grantsToMe.errMsg = action.error.message;
+        state.grantsToMe[action.meta.arg.chainID].status = "rejected";
```
Comment (aleem1314) [inline frontend/src/features/authz/authzSlice.js:475]: Create a const for `action.meta.arg.chainID`.

File: frontend/src/features/authz/authzSlice.js [hunk 3]
```diff
@@ -17,10 +17,10 @@ import { signAndBroadcast } from "../../utils/signing";
 
 const initialState = {
   grantsToMe: {
-    status: "idle",
-    errMsg: "",
-    grants: [],
-    pagination: {},
+    // status: "idle",
```
Comment (aleem1314) [inline frontend/src/features/authz/authzSlice.js:20]: Why this code is commented?

File: frontend/src/components/Send.js [hunk 4]
```diff
@@ -18,16 +22,26 @@ Send.propTypes = {
...
+    setGranter,
+    granter
+  } = props;
 
-  const params = useParams();
-
-  const selectNetwork = useSelector(
-    (state) => state.common.selectedNetwork.chainName
-  );
+  // const [granter, setGranter] = React.useState(
```
Comment (aleem1314) [inline frontend/src/components/Send.js:42]: Why this code is commented?

File: frontend/src/features/authz/authzSlice.js (hunk 3 above)
Comment (Hemanthghs) [reply frontend/src/features/authz/authzSlice.js:20]: We are adding status, errMsg, grants, and pagination for each chain in the reducer. So, these are not used now.
I will remove these comments.

File: frontend/src/components/Send.js (hunk 4 above)
Comment (Hemanthghs) [reply frontend/src/components/Send.js:42]: We are passing granter and setGranter as props to Send.js.
I will remove these comments.

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**